api/v1alpha1/thanosquery_types.go

```diff
@@ -23,22 +23,28 @@ import (
 // ThanosQuerySpec defines the desired state of ThanosQuery
 type ThanosQuerySpec struct {
 	CommonThanosFields `json:",inline"`
-
+	// Replicas is the number of router replicas.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Required
+	Replicas int32 `json:"replicas,omitempty"`
+	// Labels are additional labels to add to the Querier component.
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
 	// Querier replica labels to configure.
-	// +optional
-	QuerierReplicaLabels []string `json:"replicaLabels,omitempty"`
-
+	// +kubebuilder:validation:Optional
```

Comment:
I wonder would a link to https://thanos.io/tip/components/query.md/#deduplication-replica-labels help here?

Reply:
Will add the flag description + link